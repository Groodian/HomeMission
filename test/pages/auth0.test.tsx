import { gql, InMemoryCache } from '@apollo/client';
import React from 'react';
import Auth0Test from '../../src/pages/auth0';
import { render } from '../testUtils';

jest.mock('next/router', () => require('next-router-mock'));

const apolloCache = new InMemoryCache();
apolloCache.writeQuery({
  query: gql`
    query User {
      user {
        id
        name
        picture
        points
      }
    }
  `,
  data: {
    user: {
      __typename: 'User',
      id: 'Baa',
      name: 'Baa',
      picture: '',
      points: 100,
    },
  },
});

describe('Auth0 test page', () => {
  it('matches snapshot', async () => {
    const { asFragment } = await render(<Auth0Test />, {
      apolloCache,
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot DE', async () => {
    const { asFragment } = await render(<Auth0Test />, {
      apolloCache,
      language: 'de',
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
