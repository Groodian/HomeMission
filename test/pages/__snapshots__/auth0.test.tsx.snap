// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth0 test page matches snapshot 1`] = `
<DocumentFragment>
  Welcome Baa! You have 100 Points. 
  .emotion-0 {
  margin: 0;
  color: #1976d2;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-color: rgba(25, 118, 210, 0.4);
}

.emotion-0:hover {
  text-decoration-color: inherit;
}

<a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways emotion-0"
    href="/api/auth/logout"
  >
    Logout
  </a>
</DocumentFragment>
`;

exports[`Auth0 test page matches snapshot DE 1`] = `
<DocumentFragment>
  Willkommen Baa! Du hast 100 Punkte. 
  .emotion-0 {
  margin: 0;
  color: #1976d2;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-color: rgba(25, 118, 210, 0.4);
}

.emotion-0:hover {
  text-decoration-color: inherit;
}

<a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways emotion-0"
    href="/api/auth/logout"
  >
    Abmelden
  </a>
</DocumentFragment>
`;
